src/windows/service/exe/LxssUserSession.cpp

```diff
@@ -1079,6 +1079,19 @@ HRESULT LxssUserSessionImpl::ExportDistribution(_In_opt_ LPCGUID DistroGuid, _In
             {
                 const wil::unique_handle userToken = wsl::windows::common::security::GetUserToken(TokenImpersonation);
                 auto runAsUser = wil::impersonate_token(userToken.get());
+
+                // Ensure the target file has the correct file extension.
+                std::wstring exportPath;
+                THROW_IF_FAILED(wil::GetFinalPathNameByHandleW(FileHandle, exportPath));
```

Comment:
We should only do that this if `FileHandle` is a file, since the user might be exporting to a pipe here